service/src/main/java/bio/terra/workspace/service/workspace/GcpCloudContextService.java

```diff
@@ -160,30 +156,27 @@ public CloudContext makeCloudContextFromDb(DbCloudContext dbCloudContext) {
   public Optional<GcpCloudContext> getGcpCloudContext(UUID workspaceUuid) {
     return workspaceDao
         .getCloudContext(workspaceUuid, CloudPlatform.GCP)
-        .flatMap(GcpCloudContext::deserialize);
+        .map(GcpCloudContext::deserialize);
   }
 
   /**
-   * Retrieve the GCP cloud context. If it does not have the policies filled in, retrieve the
-   * policies from Sam, fill them in, and update the cloud context.
-   *
-   * <p>This is used during controlled resource create. Since the caller may not have permission to
-   * read the workspace policies, we use the WSM SA to query Sam.
+   * Retrieve the GCP cloud context. This is used during controlled resource create.
    *
    * @param workspaceUuid workspace identifier of the cloud context
    * @return GCP cloud context with all policies filled in.
    */
-  public GcpCloudContext getRequiredGcpCloudContext(
-      UUID workspaceUuid, AuthenticatedUserRequest userRequest) throws InterruptedException {
-    return getGcpCloudContext(workspaceUuid)
-        .orElseThrow(
-            () -> new CloudContextRequiredException("Operation requires GCP cloud context"));
+  public GcpCloudContext getRequiredGcpCloudContext(UUID workspaceUuid) {
+    GcpCloudContext cloudContext =
+        getGcpCloudContext(workspaceUuid)
+            .orElseThrow(
+                () -> new CloudContextRequiredException("Operation requires GCP cloud context"));
+    return cloudContext;
   }
 
   /**
    * Helper method for looking up the GCP project ID for a given workspace ID, if one exists. Unlike
-   * {@link #getRequiredGcpProject(UUID)}, this returns an empty Optional instead of throwing if the
-   * given workspace does not have a GCP cloud context. NOTE: no user auth validation
+   * {@link #getRequiredGcpProject}, this returns an empty Optional instead of throwing if the given
+   * workspace does not have a GCP cloud context.
    *
    * @param workspaceUuid workspace identifier of the cloud context
    * @return optional GCP project from the cloud context
```

Comment:
If you call this method on a `CREATING` context, it will return an `Optional<String>` with value `null` instead of an empty Optional, which seems misleading to callers

Reply:
(this comment is for `Optional<String> getGcpProject(UUID workspaceUuid)`, which Github wouldn't let me comment on directly since it's unchanged)

Reply:
I didn't read the doc on `Optional.map` carefully enough, this comment is wrong